src/pcrlock/pcrlock.c

```diff
@@ -1966,6 +1966,30 @@ static int event_log_map_components(EventLog *el) {
         return event_log_validate_fully_recognized(el);
 }
 
+static int tpm2_check_and_allocate(Tpm2Context** ret_tc) {
+        _cleanup_(tpm2_context_unrefp) Tpm2Context *tc = NULL;
+        int r;
+
+        if (tpm2_support() != TPM2_SUPPORT_FULL)
+                return log_error_errno(SYNTHETIC_ERRNO(EOPNOTSUPP), "Sorry, system lacks full TPM2 support.");
+
+        r = tpm2_context_new_or_warn(/* device= */ NULL, &tc);
+        if (r < 0)
+                return log_error_errno(r, "Failed to create TPM2 context: %m");
```

Comment:
tpm2_context_new_or_warn() already log on its own, hence we don't have to log here again. (the name of the func is supposed to remind you of that)